File: backend/src/main/java/com/tobeto/hotel_reservation_pair_6/services/dtos/roomDtos/responses/GetRoomResponse.java
package com.tobeto.hotel_reservation_pair_6.services.dtos.roomDtos.responses;

import com.iyzipay.model.Currency;
import com.tobeto.hotel_reservation_pair_6.services.dtos.roomBedDtos.responses.GetRoomBedResponse;
import com.tobeto.hotel_reservation_pair_6.services.dtos.roomFeatureDtos.responses.GetRoomFeatureResponse;
import com.tobeto.hotel_reservation_pair_6.services.dtos.roomImageDtos.responses.GetRoomImageResponse;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class GetRoomResponse {

    private long id;

    private int quantity;

    private double dailyPrice;

    private Currency currency;

    private int capacity;

    private int squareMeterSize;

    private int hotelId;

    private int roomTypeId;

    private String roomTypeName;

    private List<GetRoomImageResponse> roomImages;

    private List<GetRoomBedResponse> roomBeds;

    private List<GetRoomFeatureResponse> roomFeatures;

}

File: frontend/src/app/services/room/model/GetRoomResponse.ts
export interface GetRoomResponse {
    id: number;
    hotelId: number;
    roomTypeId: number;
    roomTypeName: string;
    quantity: number;
    dailyPrice: number;
    currency: string;
    capacity: number;
    squareMeterSize: number;
    roomImages: GetRoomImageResponse[];
    roomBeds: GetRoomBedResponse[];
    roomFeatures: GetRoomFeatureResponse[];
  }
  
interface GetRoomImageResponse {
    id: number;
    url: string;
  }
  
interface GetRoomBedResponse {
    id: number;
    quantity: number;
    bedName: string;
  }
  
interface GetRoomFeatureResponse {
    id: number;
    featureName: string;
  }
